components/brave_wallet_ui/components/extension/assets-panel/index.tsx

```diff
@@ -44,6 +57,42 @@ const AssetsPanel = (props: Props) => {
     })
   }
 
+  const { currentData: selectedNetwork } = useGetSelectedChainQuery(undefined)
+  const { data: userTokens } = useGetUserTokensRegistryQuery(
+    undefined,
+    {
+      selectFromResult: (res) => ({
+        ...res,
+        data: selectedNetwork !== undefined
+          ? selectAllUserAssetsByChainIdFromQueryResult(res, selectedNetwork.chainId)
```

Comment:
This selector should be a unique instance for this component so that it doesnt overrride memoization within other components using this selector with a different chainId

Reply:
Fixed it here: https://github.com/brave/brave-core/pull/18735/commits/bcbe8fb42d69e47f94bf54df638bb98216de5ba1

Let me know if this is what you had in mind.